### ISSUE TITLE
DataList::relation does not seem to work

### ISSUE THREAD

**Issue by JorisDebonnet** [silverstripe/silverstripe-framework#8043]
## Affected Version
silverstripe/recipe-cms 1.1.0
(i.e. framework 4.1.0)

## Description

`DataList::relation` has the following example in the documentation: 
```php
// Get members from all Groups:
DataList::Create(\SilverStripe\Security\Group::class)->relation("Members");
```
But this returns an empty list. From the example's description, that should at least list the Default Administrator (who belongs to the Administrator Group). Is the method broken, or is the example bad?

(I'm trying to [fix an issue in symbiote/silverstripe-memberprofiles](https://github.com/symbiote/silverstripe-memberprofiles/pull/138#issuecomment-368744525) where I have the same issue of `relation("Members")` returning an empty list)

## Steps to Reproduce
Just try this:
```php
// Get number of Members that are linked to any Group (supposedly):
echo DataList::Create(\SilverStripe\Security\Group::class)->relation("Members")->Count(); // 0
// Get number of Members linked to the Administrator Group:
echo \SilverStripe\Security\Group::get()->byID(2)->Members()->Count(); // 1 or more
```

**Comment by maxime-rainville** [comment]
I've confirmed the bug by running @JorisDebonnet's sample code.

I've tried creating a normal user and adding the to the administrator (as opposed to using the default admin) and I still get an empty list.

I think the problems is specific to the `Members` relation somehow. If I switch to the `Roles` many-many-relation, I get the expected results.

**Comment by maxime-rainville** [comment]
Problem is probably related to the [`Groups::Member()`](https://github.com/silverstripe/silverstripe-framework/blob/4/src/Security/Group.php#L311) method. 

Groups can have SubGroups. So we override the Members relation on groups so that members of child groups get return as well.

**Comment by sminnee** [comment]
Yeah the `relation()` method assumes a "normal" relation to some extent, which Members is not.
